frontend/discourse/app/components/second-factor-input.gjs

```diff
@@ -1,56 +1,36 @@
 import Component from "@glimmer/component";
-import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { on } from "@ember/modifier";
+import DOTP from "discourse/components/d-otp";
+import withEventValue from "discourse/helpers/with-event-value";
 import { SECOND_FACTOR_METHODS } from "discourse/models/user";
+import autoFocus from "discourse/modifiers/auto-focus";
 
 export default class SecondFactorInput extends Component {
-  _focusInput(input) {
-    input.focus();
-  }
-
   get isTotp() {
     return this.args.secondFactorMethod === SECOND_FACTOR_METHODS.TOTP;
   }
 
-  get isBackupCode() {
-    return this.args.secondFactorMethod === SECOND_FACTOR_METHODS.BACKUP_CODE;
-  }
-
-  get type() {
-    if (this.isTotp) {
-      return "tel";
-    } else if (this.isBackupCode) {
-      return "text";
```

Comment:
Previously the `tel` type ensured that users could only use numbers. Should we keep that in the new implementation? Maybe as an optional parameter on the OTP component.

Reply:
No other OTP implementation I have found in the wild does this, Im not sure it's a good practice to say it's a tel, when it's not a tel.